src/twisted/runner/procmon.py

```diff
@@ -5,11 +5,36 @@
 """
 Support for starting, monitoring, and restarting child process.
 """
-from twisted.python import log
+import attr
+import incremental
+
+from twisted.python import log, deprecate
 from twisted.internet import error, protocol, reactor as _reactor
 from twisted.application import service
 from twisted.protocols import basic
 
+@attr.s(frozen=True)
+class _Process(object):
+
+    """
+    The details of a process to be restarted.
```

Comment:
I was expecting to see docstring for each member of this class .

-------

minor comment.

details seem generic... but I don't have a better name.

Maybe "parameters" ... but this can be confused with arguments .... maybe "attributes"?